tests/test_security_path_traversal.py

```
@@ -204,3 +204,35 @@ def test_run_tailer_path_validation(tmp_path):
 
     with pytest.raises(ValueError, match="run_id"):
         tailer.tail_run("..")
+
+
+def test_events_routes_path_validation(tmp_path):
+    """Test that events module routes validate path components."""
+    import asyncio
+    from swarm.api.routes.events import generate_run_events, write_event, write_event_sync
+
+    async def run_tests():
```

Ruff (0.15.5)

[warning] Missing return type annotation for private function `run_tests`

Add return type annotation: `None`

(ANN202)